File: src/modules/music.ts
// ─────────────────────────────────────────────────────────────
//  DSFlix SDK  –  src/modules/music.ts
// ─────────────────────────────────────────────────────────────

import { HttpClient } from '../http';
import { MusicResponse, MusicSearchOptions, MusicTrendingOptions } from '../types';

export class MusicModule {
  constructor(private http: HttpClient) {}

  /** Get currently trending music tracks */
  getTrending(options?: MusicTrendingOptions): Promise<MusicResponse> {
    return this.http.get('/api/v2/music/trending', options as Record<string, string | number | boolean | undefined>);
  }

  /** Search for music tracks by title or artist */
  search(query: string, options?: MusicSearchOptions): Promise<MusicResponse> {
    return this.http.get('/api/v2/music/search', { query, ...options } as Record<string, string | number | boolean | undefined>);
  }
}

File: python/dsflix_sdk/modules/music.py
from __future__ import annotations

from ..http import HttpClient


class MusicModule:
    def __init__(self, http: HttpClient) -> None:
        self.http = http

    def get_trending(self, page: int | None = None, language: str | None = None) -> dict:
        return self.http.get("/api/v2/music/trending", {"page": page, "language": language})

    def search(self, query: str, page: int | None = None, language: str | None = None) -> dict:
        return self.http.get("/api/v2/music/search", {"query": query, "page": page, "language": language})
